api/v1alpha08/sonataflowplatform_types.go

```diff
@@ -46,6 +46,57 @@ type SonataFlowPlatformSpec struct {
 	// `sonataflow.org/profile: prod`
 	// +optional
 	Services ServicesPlatformSpec `json:"services,omitempty"`
+	// Persistence defines the platform persistence configuration. When this field is set,
+	// the configuration is used as the persistence for platform services and sonataflow instances
+	// that don't provide one of their own.
+	// +optional
+	Persistence *PlatformPersistenceSpec `json:"persistence,omitempty"`
+}
+
+// PlatformPersistenceSpec configures the DataBase support for both platform services and workflows. For services, it allows
+// configuring a generic database connectivity if the service does not come with its own configured. In case of the workflow
+// the configuration is defined by the image name of the supported database. When the workflow is configured to require
+// a database backend type supported by the platform (that is, the platform CR contains an image name for that database type)
+// , the workflow pod is configured to contain a sidecar container running the database image defined here, and the workflow
+// JDBC connectivity is setup to point to the DB container.
```

Comment:
The sidecar should be in the services, right? JS/DI.

Reply:
Good catch!. This was in my initial proposal to add the postgreSQL container to the workflow for local persistence, but after discussing with @wmedvede I removed this since it made no sense.

Reply:
Done